src/segmentedfilesystem/PacketStructure.java

```diff
@@ -0,0 +1,45 @@
+package segmentedfilesystem;
+import java.net.DatagramPacket;
+import java.util.Arrays;
+
+public class PacketStructure {
+    private byte status;
+    private byte ID;
+    private byte[] pnumber = new byte[2];
+    private byte[] data;
+    private int length;
+
+    PacketManager manager = new PacketManager();
+    public PacketStructure(DatagramPacket packet){
+        status = packet.getData()[0];
+        ID = packet.getData()[1];
+        length = packet.getLength();
+        if(determineType(packet)) {
+            data = packet.getData();
+        }
+        else{
+            pnumber[0] = packet.getData()[2];
+            pnumber[1] = packet.getData()[3];
+            data = Arrays.copyOfRange(packet.getData(),4,packet.getData().length-4);
+        }
+    }
+
+    private boolean determineType(DatagramPacket packet){

```

Comment:
Not a great name. I have no idea from the name what _true_ and _false_ are supposed to mean  – I have to read the code to figure that out.